fix: bugfix and update Preview.tsx

Stijnus opened the pull request:
Fix for :  https://github.com/stackblitz-labs/bolt.diy/issues/1097 

# Device Frame Feature Implementation

## Key Features Added
1. **Device Frames for Mobile and Tablet Previews**
   - Added visual frames for iPhone and iPad devices
   - Implemented both portrait and landscape orientations
   - Created realistic device bezels with notches and home buttons

2. **Device Options Panel**
   - Added toggle for showing/hiding device frames
   - Added toggle for landscape/portrait orientation
   - Updated device size information display

3. **External Window Preview Enhancements**
   - Fixed "about:blank" issue with external previews
   - Implemented reliable window creation with proper dimensions
   - Added device name and orientation labels

## Technical Implementation Details
1. **Frame Rendering Approach**
   - Created a complete HTML document with styling for device frames
   - Used document.write() to inject content into new window
   - Implemented proper iframe loading within the device frame

2. **Responsive Design**
   - Dynamic adjustment of frame elements based on device type
   - Proper handling of orientation changes (landscape/portrait)
   - Appropriate sizing for different device types (mobile vs tablet)

3. **Security and Reliability**
   - Avoided cross-origin manipulation issues
   - Implemented error handling for window creation
   - Maintained proper sandbox attributes for security

## UI/UX Improvements
1. **Visual Enhancements**
   - Added realistic device styling (rounded corners, notches, buttons)
   - Implemented proper shadows and depth for 3D appearance
   - Created clean, minimal interface for device controls

2. **User Controls**
   - Intuitive toggles for device frame and orientation
   - Clear labeling of device dimensions and frame status
   - Maintained existing preview functionality while adding new features

## Code Quality
1. **Maintainability**
   - Used TypeScript interfaces for proper typing
   - Implemented clean separation of concerns
   - Added appropriate comments for complex sections

2. **Performance**
   - Optimized rendering approach to avoid unnecessary reflows
   - Used efficient DOM manipulation techniques
   - Maintained responsive performance across different device sizes

leex279 commented:
#1474 isnt this partly the same?

Stijnus commented:
yes, and it has some extra features also...

emilio-balda commented:
Hey @Stijnus I was trying out this pull request and was not able to resize anymore. Here is a loom:
https://www.loom.com/share/009a6afea60b4a8bbd29e47d35bcbe94?sid=220c1fb9-0d3f-47fc-81bd-67ff1fe1e8f1

Stijnus commented:
Hey, @emilio-balda thank you for the feedback i will check it out this weekend and let you know when its ready for testing.

Stijnus commented:
i have made some adjustments and all should be working now :

## Fixed preview functionality while maintaining resize capability:
- Removed `e.preventDefault()` calls in pointer event handlers that were blocking normal event flow
- Improved pointer capture release logic with proper element detection
- Added safety checks to prevent errors when releasing pointer captures

## Device Preview Improvements

### Fixed Landscape Mode Display
- Increased horizontal padding for mobile devices in landscape mode
- Adjusted frame dimensions to ensure the full device frame is visible
- Updated notch and home button positions for better alignment

### Added Visual Improvements
- Added a checkmark indicator to show the currently selected device
- Implemented dark mode support with responsive frame colors:
  - Dark gray (`#111`) for light mode
  - Medium gray (`#555`) for dark mode

### Code Improvements
- Created `getFrameColor()` function for dynamic color detection
- Added media query listeners to react to system color scheme changes
- Improved frame dimension calculations to handle different device types
- Ensured consistency between inline preview and external window



<img width="1800" alt="image" src="https://github.com/user-attachments/assets/29efa0cf-45a0-483a-b92c-bf4bfd5562bd" />

Stijnus commented:
Hey @emilio-balda i have updated the PR could you check once more?

Thx
